unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1382,3 +1383,28 @@ public class TypedVariable {
     public string type;
     public object value;
 }
+
+public class ShouldSerializeContractResolver : DefaultContractResolver
+{
+   public static readonly ShouldSerializeContractResolver Instance = new ShouldSerializeContractResolver();
+
+   protected override JsonProperty CreateProperty( MemberInfo member,
```

Comment:
Hopefully this does not make it much slower. Why did we not need this before, or did we? Is this to the net framework change and support with a particular Newtonsoft version?